rpc/src/module/ipc.rs

```diff
@@ -0,0 +1,784 @@
+                },
+                &env.script_hash.0.pack(),
+            )
+            .ok_or_else(|| {
+                RPCError::custom_with_error(
+                    RPCError::IPC,
+                    format!(
+                        "{:?}",
+                        ckb_script::ScriptError::ScriptNotFound(env.script_hash.0.pack())
+                    ),
+                )
+            })?;
+        let ci = data
+            .extract_referenced_dep_index(&sg.script)
+            .map_err(|e| RPCError::custom_with_error(RPCError::IPC, e))?;
+        let mut meta = tx.resolved_cell_deps[ci].clone();
+        meta.mem_cell_data = Some(ipc_script_data);
+        dl.cell
+            .insert(meta.out_point.clone(), CellStatus::Live(meta));
+        return Ok(env);
+    }
+
+    fn ipc_env(
+        &self,
+        dl: &mut Resource,
+        ipc_script_data: Bytes,
+        env: Option<IpcEnv>,
+    ) -> Result<IpcEnv> {
+        match env {
+            Some(val) => self.ipc_env_some(dl, ipc_script_data, val),
+            None => self.ipc_env_none(dl, ipc_script_data),
+        }
+    }
+
+    fn ipc_script_data_by_out_point(&self, script_locator: IpcScriptLocator) -> Result<Bytes> {
+        let out_point = script_locator.out_point.unwrap();
+        let out_point: packed::OutPoint = out_point.into();
+        if let Some((data, _)) = self.shared.snapshot().get_cell_data(&out_point) {
+            Ok(data)
+        } else {
+            Err(RPCError::custom(
+                RPCError::IPC,
+                format!("Get out point failed: {}", out_point),
+            ))
+        }
+    }
+
+    fn ipc_script_data_by_type_id_args(&self, script_locator: IpcScriptLocator) -> Result<Bytes> {
+        match &self.indexer_rpc_impl {
+            Some(data) => {
+                let search_key = IndexerSearchKey {
+                    script: Script {
+                        code_hash: TYPE_ID_CODE_HASH,
+                        hash_type: ScriptHashType::Type,
+                        args: JsonBytes::from_vec(
+                            script_locator.type_id_args.unwrap().as_bytes().to_vec(),
+                        ),
+                    },
+                    script_type: IndexerScriptType::Type,
+                    ..Default::default()
+                };
+                let object_vec = data
+                    .handle
+                    .get_cells(search_key, IndexerOrder::Desc, 1.into(), None)
+                    .map_err(|e| RPCError::custom_with_error(RPCError::IPC, e))?
+                    .objects;
+                Ok(object_vec[0]
+                    .output_data
+                    .clone()
+                    .ok_or_else(|| RPCError::custom(RPCError::IPC, "Unexpected"))
+                    .map(|e| e.into_bytes())?)
+            }
+            None => Err(RPCError::custom(
+                RPCError::IPCIndexerIsDisabled,
+                "Query by type id requires enabling indexer",
```

Comment:
`                "Query by type id requires enabling Indexer in `[rpc.modules]`",`